src/main/java/se/sundsvall/emailreader/service/EmailScheduler.java

```diff
@@ -1,97 +1,51 @@
 package se.sundsvall.emailreader.service;
 
-
-import java.text.MessageFormat;
-import java.time.OffsetDateTime;
-
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Component;
 
-import microsoft.exchange.webservices.data.property.complex.ItemId;
 import net.javacrumbs.shedlock.spring.annotation.SchedulerLock;
-import se.sundsvall.emailreader.integration.db.CredentialsRepository;
-import se.sundsvall.emailreader.integration.db.EmailRepository;
-import se.sundsvall.emailreader.integration.db.entity.EmailEntity;
-import se.sundsvall.emailreader.integration.ews.EWSIntegration;
-import se.sundsvall.emailreader.integration.messaging.MessagingIntegration;
-import se.sundsvall.emailreader.service.mapper.EmailMapper;
-import se.sundsvall.emailreader.utility.EncryptionUtility;
 
 @Component
 public class EmailScheduler {
 
-	private static final String EMAIL_SUBJECT = "[Warning] EmailReader has detected unhandled emails";
-
 	private static final Logger log = LoggerFactory.getLogger(EmailScheduler.class);
 
-	private final EWSIntegration ewsIntegration;
-
-	private final MessagingIntegration messagingIntegration;
-
-	private final EmailRepository emailRepository;
-
-	private final CredentialsRepository credentialsRepository;
+	private final EmailService emailService;
 
-	private final EncryptionUtility encryptionUtility;
-
-
-	public EmailScheduler(final EWSIntegration ewsIntegration, final EmailRepository emailRepository,
-		final CredentialsRepository credentialsRepository, final EncryptionUtility encryptionUtility, final MessagingIntegration messagingIntegration) {
-
-		this.ewsIntegration = ewsIntegration;
-		this.emailRepository = emailRepository;
-		this.credentialsRepository = credentialsRepository;
-		this.encryptionUtility = encryptionUtility;
-		this.messagingIntegration = messagingIntegration;
+	public EmailScheduler(final EmailService emailService) {
+		this.emailService = emailService;
 	}
 
+	/**
+	 * This method is scheduled to run at a fixed rate to check for new emails and save them to the
+	 * database. It is also responsible for moving the emails to the processed folder.
+	 */
 	@Scheduled(initialDelayString = "${scheduled.initial-delay}", fixedRateString = "${scheduled.fixed-rate}")
```

Comment:
Is it not better to use cron-expression instead of fixed-rate when we have multiple instances that executes the job? I.e. if scheduled at at fixed rate, the time is "initialized" from when the service starts. If instance A starts X instants before instance B, there will be two executions, one for instance A and one for instance B (where execution from instance B is probably unnecessary).

If the scheduler(s) use cron for every hour, there will only be one execution every hour (as schedulerLock will prevent one of the instances to execute).